terraform/modules/gro_ingestion_service/lambdas/src/functions/insertXml.ts

```diff
@@ -0,0 +1,127 @@
+const dateOfBirthOptions = ["1912-02-29", ""]
+const dateOfDeathOptions = ["2012-02-29", ""]
+const genderOptions = ["1", "2"]
+const addressOptions = [
+    ["10 Test Street", "Test Town", "Test County"],
+    ["", "Westminster", "London"],
+    ["27 Brown Lane", "", "Somerset"],
+    ["65 Link Road", "Southport", ""],
+]
+const postcodeOptions = ["PR8 1HY", ""]
+
+const handler: Handler = async (event): Promise<InsertXmlResponse> => {
+    const numberOfRecords =
+        event["detail-type"] === "Scheduled Event" || !event["numberOfRecords"] ? 25 : event["numberOfRecords"]
+    const fileKey = `${new Date().toISOString()}-fake-gro.xml`
+    const logParams: { fileKey: string; error?: Error } = { fileKey: fileKey }
+
+    try {
+        const xml = generateXml(numberOfRecords)
+        await uploadXml(xml, fileKey)
+        console.log("Successfully generated GRO file", logParams)
+        return {
+            statusCode: 200,
+            payload: fileKey,
+        }
+    } catch (err) {
+        logParams.error = err
+        console.error("Failed to generate GRO file", logParams)
+        return {
+            statusCode: 500,
+        }
+    }
+}
+
+const uploadXml = async (xml: string, fileKey: string) => {
+    const putObjectCommand = new PutObjectCommand({
+        Bucket: config.s3BucketArn,
+        Key: fileKey,
+        Body: xml,
+    })
+    await s3Client.send(putObjectCommand)
+}
+
+const generateXml = (numberOfRecords: number) => {
+    const deathRecords = [...Array(numberOfRecords)].map(_ => createDeathRecord())
+    return xmlBuilder.build({
+        DeathRegistrationGroup: {
+            DeathRegistration: deathRecords,
+            RecordCount: numberOfRecords,
+        },
+    })
+}
+
+const createDeathRecord = () => {
+    const title = getRandomElement(titleOptions)
+    const forename = getRandomElement(forenameOptions)
+    const surname = getRandomElement(surnameOptions)
+    const date = new Date().toISOString().slice(0, 19)
+
+    return {
+        DeathRegistration: {
+            RegistrationId: randomUUID(),
+            RecordLockedDateTime: date,
+            RecordUpdateDateTime: date,
+            RecordUpdateReason: "",
+            DeceasedName: {
+                PersonNameTitle: title,
+                PersonGivenName: forename,
+                PersonFamilyName: surname,
+                PersonNameSuffix: "",
+            },
+            DeceasedAliasName: {
+                PersonNameTitle: title,
+                PersonGivenName: forename,
+                PersonFamilyName: surname,
+                PersonNameSuffix: "",
+            },
+            DeceasedAliasNameType: "",
+            DeceasedMaidenNameType: "",
+            DeceasedGender: getRandomElement(genderOptions),
+            DeceasedDeathDate: {
+                PersonDeathDate: getRandomElement(dateOfDeathOptions),
+                VerificationLevel: "03",
+            },
+            PartialMonthOfDeath: "",
```

Comment:
Should we be populating these fields from the DeathDate?

Reply:
I filled in the sections that were filled in in the example schema - [here](https://docs.google.com/spreadsheets/d/1fFMlFL52P2FYAd5X-0PlGbKuPn-4uEkn/edit#gid=1526528896) someone seemed to ask if these should be calculated, but that seems to be on the DWP end?

Reply:
So at the moment we are passing this through so I would suggest adding it, once we get the sample JSON back from DWP we will have to update all our models so thats ok to update then